content/en/docs/contribute/style/diagram-guide.md

```diff
@@ -0,0 +1,577 @@
+```
+classDef k8s fill:#326ce5,stroke:#fff,stroke-width:4px,color:#fff; // defines style for the k8s class
+class ingress,service,pod1,pod2 k8s; // k8s class is applied to elements ingress, service, pod1 and pod2.
+```
+You can include one or multiple `classDef` and `class` statements in your diagram. You can also use the official K8s `#326ce5` hex color code for K8s components in your diagram.
+
+For more information on styling and classes, see [Mermaid Styling and classes docs](https://mermaid-js.github.io/mermaid/#/flowchart?id=styling-and-classes).
+
+## How to use captions
+
+A caption is a brief description of a diagram. A title or a short description of the diagram are examples of captions. Captions aren't meant to replace explanatory text you have in your documentation. Rather, they serve as a "context link" between that text and your diagram. 
+
+The combination of some text and a diagram tied together with a caption provide a concise representation of the information you wish to convey to the user. 
+
+Without captions, you are asking the user to scan the text above or below the diagram to figure out a meaning. This can be frustrating for the user. 
+
+{{< note >}}
+You should always add a caption to each diagram in your documentation.
+{{< /note >}}
+
+Proper captioning involves three components: diagram, diagram caption and the diagram referral. As an example, consider a diagram `.svg` image file saved to `/images/docs/components-of-kubernetes.svg`.
+
+**Diagram**
+
+Here is the `{{</* figure */>}}` shortcode for the diagram:
+```text
+{{</* figure src="/images/docs/components-of-kubernetes.svg" alt="Kubernetes pod running inside a cluster" class="diagram-large" */>}}
+```
+You should only pass the `src`, `alt` and `class` values into the `{{</* figure */>}}` shortcode. You can adjust the size of the diagram using `diagram-large`, `diagram-medium` and `diagram-small` classes. You can't style the `caption` value in this shortcode, and it only supports diagram image files. 
+
+{{< note >}}
+Diagrams created using the `Inline` method don't use the `{{</* figure */>}}` shortcode.
+{{< /note >}}
+
+**Diagram Caption**
+
+Next, add a diagram caption. You must use the following format:
+```text
+<figcaption><p style="padding-left: 310px; font-weight: bold">Figure #. CAPTION TEXT</p></figcaption><br>
+```
+Here are several items to consider when adding diagram captions: 
+
+* Use`<figcation></figcaption>` tags to add a diagram caption for `Inline`, `Mermaid+SVG` and `External Tool` diagrams. 
+* Wrap caption text inside `<p></p>` tags using `style="padding-left: 310px; font-weight: bold"`. This provides a consistent style for the caption text.
+* Prepend your diagram caption with `Figure NUMBER.`. You must use `Figure` and the number must be unique for each diagram in your documentation page. 
```

Comment:
This is the sort of thing that Hugo can compute for us, if we want it to. Let's not make authors do the work here.